addons/mail/models/mail_link_preview.py

```diff
@@ -38,16 +39,24 @@ def _create_from_message_and_notify(self, message):
             'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'
         })
         link_preview_values = []
+        existing_link_preview_ids = []
+        existing_link_previews_by_url = dict((preview.source_url, preview) for preview in self.search([('message_id', '=', message.id)]))
         for url in list(urls):
+            if url in existing_link_previews_by_url:
+                preview = existing_link_previews_by_url.pop(url)
```

Comment:
it gets kind of confusing with having initially existing_link_previews_by_url, then using pop, and then it becomes previews to delete (but the variable name hasn't changed)

Reply:
the idea was to reuse the variable as much as possible so that we don't do extra assigned value or filters. But indeed the readability was bad, I changed the code so that it is better to use.